streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -611,6 +607,9 @@ boolean runLoop() {
                 if (processingMode == ProcessingMode.EXACTLY_ONCE_ALPHA || processingMode == ProcessingMode.EXACTLY_ONCE_BETA) {
                     return false;
                 }
+                // TODO: extract this catch block to encompass all of runLoop, in case exceptions slip through one of the
```

Comment:
I tried doing this here to see what would happen, and it caused `StreamsUncaughtExceptionHandlerIntegrationTest#shouldShutdownSingleThreadApplication` to fail. I backed off from it in this PR but @wcarlson5, any insight into why that might be?

Reply:
I would think that https://github.com/apache/kafka/pull/10387 should fix that by adding a thread. Because the loop needs to turn over one more time to trigger the shutdown rebalance.

Reply:
Ah, gotcha. Maybe you can tack this onto your PR if you think it makes sense, otherwise I can file a quick ticket to follow up on it

Reply:
I am not sure that extracting this all is the right thing to do. But why don't you make a ticket and we can come back to it

Reply:
If we don't extract it then anything thrown in any of these catch blocks (eg from `handleCorruption`) will miss the exception handler, not be recorded in the failed thread sensor, etc. I ended up just opening a quick hotfix PR so we can get this into any ongoing releases that need it without coupling it to anything else we have going on right now: https://github.com/apache/kafka/pull/10417